src/features/annotation-thread/AnnotationThreadContent.js

```diff
@@ -0,0 +1,180 @@
+import type { BoxItemPermission, SelectorItems } from '../../common/types/core';
+
+type Props = {
+    annotationId: string,
+    api: API,
+    fileId: string,
+    filePermissions: BoxItemPermission,
+    getCollaboratorsWithQuery: Function,
+    getUserProfileUrl?: GetProfileUrlCallback,
+    onError: (error: ElementsXhrError | Error, code: string, contextInfo?: Object, origin?: ElementOrigin) => void,
+};
+
+const AnnotationThreadContent = ({
+    annotationId,
+    api,
+    fileId,
+    getCollaboratorsWithQuery,
+    getUserProfileUrl,
+    filePermissions,
+    onError,
+}: Props) => {
+    const [mentionSelectorContacts, setMentionSelectorContacts] = React.useState();
+    const [annotation, setAnnotation] = React.useState();
+    const [annotationError, setAnnotationError] = React.useState();
+    const [isLoading, setIsLoading] = React.useState(true);
+
+    const getAnnotationSuccess = (fetchedAnnotation: Annotation): void => {
+        setAnnotation(fetchedAnnotation);
+        setAnnotationError(undefined);
+        setIsLoading(false);
+    };
+
+    const errorCallback = React.useCallback(
+        (error: ElementsXhrError, code: string, contextInfo: Object = {}): void => {
+            /* eslint-disable no-console */
+            console.error(error);
+            /* eslint-enable no-console */
+            onError && onError(error, code, contextInfo);
+        },
+        [onError],
+    );
+
+    const annotationErrorCallback = React.useCallback(
+        (e: ElementsXhrError, code: string) => {
+            setAnnotation(undefined);
+            setIsLoading(false);
+            setAnnotationError({
+                maskError: {
+                    errorHeader: commonMessages.defaultErrorMaskHeaderMessage,
+                    errorSubHeader: commonMessages.defaultErrorMaskSubHeaderMessage,
+                },
+            });
+
+            errorCallback(e, code, {
+                error: e,
+            });
+        },
+        [errorCallback],
+    );
+
+    const fetchAnnotation = React.useCallback(() => {
+        api.getAnnotationsAPI(false).getAnnotation(
+            fileId,
+            annotationId,
+            filePermissions,
+            getAnnotationSuccess,
+            annotationErrorCallback,
+            true,
+        );
+    }, [annotationErrorCallback, annotationId, api, fileId, filePermissions]);
+
+    React.useEffect(() => {
+        fetchAnnotation();
+    }, [annotationId, fetchAnnotation]);
+
+    const annotationSuccessCallback = () => {
+        setAnnotation({ ...annotation, isPendind: false });
+        fetchAnnotation();
+    };
+
+    const annotationDeleteSuccessCallback = () => {
+        setAnnotation({ ...annotation, isPendind: false });
+    };
+
+    const handleAnnotationDelete = ({ id, permissions }: { id: string, permissions: AnnotationPermission }) => {
+        setAnnotation({ ...annotation, isPendind: true });
```

Comment:
Seems like it's possible to have both the `Sidebar` and `AnnotationThread` components open at the same time. What is the expected experience when any of these delete/edit/resolve annotations actions are made? Will the sidebar also reflect a pending state for that given annotation? The Feed api manages it's own pending state which is separate from the state being managed here

Reply:
Good call, I will consider this case while creating event bus for sharing state between Sidebar and AnnotationThread